src/MailHandling/SendNewPasswordMail.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\MailHandling;
+
+use App\Entity\User;
+use App\Entity\UserResetPasswordToken;
+use App\Repository\UserResetPasswordTokenRepository;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Uid\Uuid;
+
+class SendNewPasswordMail implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private UserResetPasswordTokenRepository $userResetPasswordTokenRepository;
+
+    private MailerInterface $mailer;
+
+    private string $passwordResetExpirationMinutes;
+
+    public function __construct(
+        UserResetPasswordTokenRepository $userResetPasswordTokenRepository,
+        MailerInterface $mailer,
+        $passwordResetExpirationMinutes
+    ) {
+        $this->userResetPasswordTokenRepository = $userResetPasswordTokenRepository;
+        $this->mailer = $mailer;
+        $this->passwordResetExpirationMinutes = $passwordResetExpirationMinutes;
+    }
+
+    public function handle(User $user): void
+    {
+        $userToken = $this->userResetPasswordTokenRepository->findOneBy(['user' => $user]);
```

Comment:
If you reached this step then the $user does NOT have a valid token generated. If it has, the validation in the controller should catch that. Therefore, you don't need this step here. This class has the responsibility to send an email to the user. It currently does that but it also generates a token and stores in the database. You are breaking the Single Responsibility Principle.